src/DataStructures/Tensor/Expressions/Magnitude.hpp

```diff
@@ -0,0 +1,63 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines functions euclidean_magnitude and magnitude
+
+#pragma once
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+
+/*!
+ * \ingroup Tensor
+ * \brief compute the Euclidean magnitude of a rank-1 tensor
+ *
+ * \returns the Euclidean magnitude of the rank-1 tensor
+ *
+ * \details
+ * Computes the square root of the sum of the squares of the components of
+ * the rank-1 tensor.
+ *
+ */
+template <typename Index>
+DataVector euclidean_magnitude(
```

Comment:
Would it be worth keeping the euclidean prefix to prevent misuse of magnitude? Somebody might actually want the magnitude with respect to a metric but forget to place the argument and the code will run.